fe/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java

```diff
@@ -1438,4 +1457,19 @@ public void cancel(CancelStmt stmt) throws DdlException {
             jobDone(schemaChangeJob);
         }
     }
+
+    private void processAddIndex(CreateIndexClause alterClause, List<Index> indexes) {
+        if (alterClause.getIndex() != null) {
+            indexes.add(alterClause.getIndex());
+        }
+    }
+
+    private void processDropIndex(DropIndexClause alterClause, List<Index> indexes) {
+        for (Index idx : indexes) {
```

Comment:
This is not a good idea to iterate list in this way when element will be deleted in the loop block